lib/set_builder/modifier/arguments.rb

```diff
@@ -0,0 +1,50 @@
+require "set_builder/parser"
+
+module SetBuilder
+  module Modifier
+    class Arguments
+      attr_reader :arguments, :expression, :tokens
+
+      include Parser
+
+      def initialize(expression)
+        @expression = expression.respond_to?(:each) ? map_legacy_arguments(expression) : expression
+        @tokens = parse(@expression)
+      end
+
+      def arity
+        list.count
+      end
+
+      def list
+        @types ||= @tokens
+          .select { |token, _| token == :arg }
+          .map { |_, type| type }
+      end
```

Comment:
You store this in the ivar `@types` — is that actually a better name for this method than `list`?